### PR TITLE
Thank you Part 1 - Thank you dialog

### PR DESCRIPTION
**Pull request by tidy-dev**
## Description

Adds the dialog for displaying the thank you message, release notes, confetti.

Does not yet invoke it - breaking work into a couple PRs for easier review.

## Release notes
Notes: no-notes

### CODE REVIEW

File: app/src/ui/thank-you/thank-you.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+import { Dialog, DialogContent } from '../dialog'
+
+import { RichText } from '../lib/rich-text'
+import { Repository } from '../../models/repository'
+import { getDotComAPIEndpoint } from '../../lib/api'
+import { GitHubRepository } from '../../models/github-repository'
+import { Owner } from '../../models/owner'
+
+// HACK: This is needed because the `Rich`Text` component
```
Comment (tidy-dev) [inline app/src/ui/thank-you/thank-you.tsx]: Directly copied form the release note component - tho this is noted as a Hack. It seems there isn't a better alternative.

File: app/src/ui/thank-you/thank-you.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    return (
+      <div className="section">
+        <ul className="entries">{options}</ul>
+      </div>
+    )
+  }
+
+  private renderConfetti(): JSX.Element | null {
```
Comment (tidy-dev) [inline app/src/ui/thank-you/thank-you.tsx:49]: I am not 100% sure. But, we would likely see better performance if we did this outside the renderer. Grab a div ref and pump the confetti in. Currently, if user moves their mouse during confetti, there can sometimes be a noticeable lag. 

Edit.. tried didn't help.. not sure what is causing the mouse move/lag thing unless there is a mouse move event trigger taking priority... oddly when running the performance profiler it behaves. 😛

Edit again: I cut the confetti down to 750 pieces in last PR and performance is much better and still looks good.

File: app/src/ui/thank-you/thank-you.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+)
+
+const ReleaseNoteHeaderLeftUri = encodePathAsUrl(
+  __dirname,
+  'static/release-note-header-left.svg'
+)
+const ReleaseNoteHeaderRightUri = encodePathAsUrl(
+  __dirname,
+  'static/release-note-header-right.svg'
+)
```
Comment (say25) [inline app/src/ui/thank-you/thank-you.tsx]: Would it be weird if we extracted this (duplicated in `release-notes-dialog.tsx`) to a common place?

Comment (sergiou87) [reply app/src/ui/thank-you/thank-you.tsx]: Yeah I agree that'd be nice if possible! 👍

**Review by sergiou87 — COMMENTED** [review]
Looks great overall!! Please consider @say25 suggestion, and :shipit:

File: app/src/ui/thank-you/thank-you.tsx (hunk 3 above)
Comment (tidy-dev) [reply app/src/ui/thank-you/thank-you.tsx]: Yep. Good point!

**Review by sergiou87 — APPROVED** [review]
Looks perfect, thanks for all those changes!! 🙏 🙇‍♂️ :shipit:

**Pull request merged**